test/e2e/service_latency.go

```diff
@@ -144,58 +165,155 @@ func runServiceLatencies(f *Framework, inParallel, total int) (output []time.Dur
 	return output, nil
 }
 
-func singleServiceLatency(f *Framework, i int) (time.Duration, error) {
-	// Make an RC with a single pod.
-	cfg := RCConfig{
-		Client:       f.Client,
-		Image:        "gcr.io/google_containers/pause:1.0",
-		Name:         fmt.Sprintf("trial-%v", i),
-		Namespace:    f.Namespace.Name,
-		Replicas:     1,
-		PollInterval: time.Second,
+type endpointQuery struct {
+	endpointsName string
+	endpoints     *api.Endpoints
+	result        chan<- struct{}
+}
+
+type endpointQueries struct {
+	requests map[string]*endpointQuery
+
+	stop        chan struct{}
+	requestChan chan *endpointQuery
+	seenChan    chan *api.Endpoints
+}
+
+func newQuerier() *endpointQueries {
+	eq := &endpointQueries{
+		requests: map[string]*endpointQuery{},
+
+		stop:        make(chan struct{}, 100),
+		requestChan: make(chan *endpointQuery),
+		seenChan:    make(chan *api.Endpoints, 100),
 	}
-	if err := RunRC(cfg); err != nil {
-		return 0, err
+	go eq.join()
```

Comment:
This is just to check for failures on the above make's, right?  Probably worth a comment explaining.